ts/nni_manager/training_service/local/localTrainingService.ts

```diff
@@ -436,7 +436,7 @@ class LocalTrainingService implements TrainingService {
 
     private checkSpecifiedGpuIndices(): void {
         const gpuCount: number = this.gpuScheduler.getSystemGpuCount();
-        if (this.designatedGpuIndices !== undefined) {
+        if (this.designatedGpuIndices !== undefined && gpuCount > 0) {
```

Comment:
What will happen if gpu count IS 0?

Reply:
Only detect gpu count when user set gpuNum>0 in config file. If gpu count is 0 in the machine, the trial job will always be waitting. We could not guarantee the detected 0 is a valid number in gpuScheduler component, then should not validate gpuIndices when count is 0.

Reply:
I suggest to return undefined or null before first detection. `getSystemGpuCount()` is only used here so I guess it won't be too much effort.

Reply:
agree, updated.